### DISCUSSION TITLE
Rotating a body using the Chipmunk library

### DISCUSSION THREAD

**Discussion by musjj** [hajimehoshi/ebiten#2789] [Q&A] [answered]
What's the idiomatic way of rotating a body with [Chipmunk](https://github.com/jakecoffman/cp)?

I'm doing something like this:

```golang
shape.Body().SetAngle(0.78)
shape.Body().SetVelocity(0, 50)
```

But the body still moves perfectly vertically. I realize that I can probably calculate the vector of the velocity to achieve a specific angle, but I'm wondering what's `SetAngle` is for if it doesn't affect the direction of the velocity?

Some examples would be appreciated.

**Comment by hajimehoshi** [comment]
I think this is not a question about Ebitengine. Please ask the author of the Chipmunk repository. Thanks,

**Reply by musjj** [reply]
Sorry about that, I couldn't find a decent community for Chipmunk so I decided to ask here (since Chipmunk seems semi-popular among Golang and Ebiten users).

**Comment by musjj** [comment, marked as answer]
Figured it out! The solution was to rotate the velocity vector after setting the angle:

```go
shape.Body().SetVelocity(0, 50)

angle := shape.Body().Angle()
inc = 0.05
shape.Body().SetAngle(angle + inc)

velocity := p.shape.Body().Velocity()
other := cp.ForAngle(inc)
shape.Body().SetVelocityVector(velocity.Rotate(other))
```
